### PR TITLE
Test create book

### PR DESCRIPTION
**Pull request by lcpulzone**
## Changes

Add note to remember change needed, add destroy method
Add to destroy method
Add remove_book method
Uncomment out delete route
Create asset for stub response
Add beginning of test for remove book
Finish testing for remove book method
Add initial method to remove a book
Update destroy action with proper file name
Fix indentation on remove test, add in test data for create book test
Added shelf_id argument, commented out finding shelf id method
Add Add in shelf_id param to add_book method in create action
Remove the remove_a_book method

## Special review requests
Forgot to switch branches before I started messing with the delete functionality!  I added in a test for the create method in the service from google books API, we should take a look at that and align on what the test should be looking for.  Added in a shelf_id param as the id's for the three shelves we're tracking never change, can discuss/explain further!

## Checklist
- [ ] Tests written where required 
- [x] All tests passing 
- [ ] SimpleCov at 100% - 97.71%
- [ ] RuboCop complaints resolved - did not run

closes #

### CODE REVIEW

File: app/controllers/user_books_controller.rb [hunk 1]
```diff
@@ -1,12 +1,18 @@
 class UserBooksController < ApplicationController
 
   def index
-    books = UserBooksFacade.all_books(params[:auth_token])
+    books = UserBooksFacade.all_books(params[:auth_token])   # needs a user id and a filter_by(can be default nothing) in the params
```
Comment (aliyamerali) [inline app/controllers/user_books_controller.rb:4]: I _think_ Lawrence mentioned that the auth_token includes user information we need to hit the Google Books API - what are you planning to use the user_id info for?

File: app/services/book_service.rb [hunk 2]
```diff
@@ -27,13 +27,16 @@ def self.books_on_shelf(shelf, auth_token)
...
-    shelf =  BookService.book_shelves(auth_token)[:items].find do |shelf|
-                shelf[:title] == "To read"
-              end
-    shelf_id = shelf[:id]
+  def self.add_book(shelf_id, volume_id, auth_token)
+    # shelf =  BookService.book_shelves(auth_token)[:items].find do |shelf|
+    #             shelf[:title] == "To read"
+    #           end
+    # shelf_id = shelf[:id]
+    conn(auth_token).get "/books/v1/mylibrary/bookshelves/#{shelf_id}/addVolume?volumeId=#{volume_id}"
```
Comment (aliyamerali) [inline app/services/book_service.rb:35]: Love this change - It really cleans things up and I love the added functionality it provides the user. Thanks for uncovering this!

File: spec/services/book_services_spec.rb [hunk 3]
```diff
@@ -86,15 +86,52 @@
...
-        #
-        # VCR.turn_off!
-        #   actual = BookService.add_book(volume_id, 'auth_token')
-        # VCR.turn_on!
+        VCR.turn_off!
+          actual = BookService.add_book(shelf_id, volume_id, 'auth_token')
+        VCR.turn_on!
+
+        expect(actual.status).to eq(200)
+        expect(actual.body).to eq("{}")
```
Comment (aliyamerali) [inline spec/services/book_services_spec.rb:109]: Thank you for adding these tests in! This makes sense to me - testing just that the response is successful with an empty body. I think I struggle fundamentally with stubbing out the service specs because it feels like we're not really testing anything  🤷‍♀️ .

**Review by aliyamerali — APPROVED** [review]
Left one or two questions but nothing blocking- looking great!
